sdk/src/main/java/io/opentelemetry/sdk/trace/SpanBuilderSdk.java

```diff
@@ -169,11 +167,24 @@
   @Override
   public Span.Builder setAttribute(String key, AttributeValue value) {
     Utils.checkNotNull(key, "key");
-    Utils.checkNotNull(value, "value");
-    if (value.getType() == Type.STRING && StringUtils.isNullOrEmpty(value.getStringValue())) {
-      return this;
+    boolean remove = true;
+    if (value != null) {
+      switch (value.getType()) {
+        case STRING:
+          remove = value.getStringValue() == null;
+          break;
+        case BOOLEAN:
+        case LONG:
+        case DOUBLE:
+          remove = false;
+          break;
+      }
+    }
```

Comment:
let's extract this logic to determine `remove` into its own method.

Reply:
speaking of which, couldn't it just be reduced to 
`(value != null) && (value.getType() == STRING) && (value.getStringValue() == null)` 
?

Reply:
Lets definitely extract the logic into its own method.

Reply:
> speaking of which, couldn't it just be reduced to
> `(value != null) && (value.getType() == STRING) && (value.getStringValue() == null)`
> ?

In vision of Array types, I prefer a switch statement.